web/docs/language/features.md

```diff
@@ -412,8 +422,113 @@ import sayHi from '@wasp/actions/sayHi'
+    - `updateQuery` (required) - The function used to perform the optimistic update. Wasp will call it with the following arguments:
+      - `item` - The argument you pass into the decorated Action.
+      - `oldData` - The currently cached value for the query identified by the specifier.
+
+Here's an example showing how to configure the Action from the previous example to perform an optimistic update: 
+```jsx {3,9,10,11,12,13,14,15,16,27} title=pages/Task.js
+import React from 'react'
+import { useQuery } from '@wasp/queries'
+import { useAction } from '@wasp/actions'
+import fetchTask from '@wasp/queries/fetchTask'
+import updateTaskIsDone from '@wasp/actions/updateTaskIsDone'
+
+const TaskPage = ({ id }) => {
+  const { data: task } = useQuery(fetchTask, { id })
+  const updateTaskIsDoneOptimistically = useAction(updateTaskIsDone, {
+    optimisticUpdates: [
+      {
+        getQuerySpecifier: ({ id }) => [fetchTask, { id }],
+        updateQuery: ({ isDone }, oldData) => ({ ...oldData, isDone })
+      }
+    ]
+  })
+
+  if (!task) {
+    return <h1>"Loading"</h1>
+  }
+
+  const { description, isDone } = task
+  return (
+    <div>
+      <p><strong>Description: </strong>{description}</p>
+      <p><strong>Is done: </strong>{isDone ? 'Yes' : 'No'}</p>
+      <button onClick={() => updateTaskIsDoneOptimistically({ id, isDone: !isDone })}>
+        Mark as {task.isDone ? 'undone' : 'done'}
+      </button>
+      <div>
+        <Link to="/">Back to main page</Link>
+      </div>
+    </div>
+  )
+}
+
+export default TaskPage 
+```
+The `useAction` hook currently only supports specifying optimistic updates. You can expect more features in future versions of Wasp.
+
+Wasp's optimistic update API is deliberately small and focuses exclusively on updating Query caches (as that's the most common use case). You might need an API that offers more options or a higher level of control. If that's the case, instead of using Wasp's `useAction` hook, you can use _react-query_'s `useMutation` hook and directly work with [their low-level API](https://tanstack.com/query/v4/docs/guides/optimistic-updates?from=reactQueryV3&original=https://react-query-v3.tanstack.com/guides/optimistic-updates).
+
+If you decide to use _react-query_'s API directly, you will need access to the Query's cache key. Wasp internally uses this key but abstracts it from the programmer. Still, you can easily obtain it by accessing the `queryCacheKey` property on a Query:
+```js
+import { fetchTasks } from '@wasp/queries'
+
+const queryKey = fetchTasks.queryCacheKey
+```
 
 ### Cache Invalidation
```

Comment:
Is there anything we might want to update in this section? Maybe mention optimistic updates in some context? You will know better, just asking.

Reply:
Definitely, forgot about this part. I'll add it in.